Ошибка при сборке osmconvert

В файле omim/tools/unix/generate_planet.sh нужно 
```bash
 [ ! -x "$OSMCTOOLS/osmconvert" ] && cc -x c -lz -O3 "$OMIM_PATH/tools/osmctools/osmconvert.c" -o "$OSMCTOOLS/osmconvert" 
``` 
заменить на 
```bash
[ ! -x "$OSMCTOOLS/osmconvert" ] && cc -x c -O3 "$OMIM_PATH/tools/osmctools/osmconvert.c" -o "$OSMCTOOLS/osmconvert" -lz
```
так как ld опции при сборке должны идти в конце.

Иначе будет ошибка: 
```bash
+ cc -x c -lz -O3 /omim/tools/osmctools/osmconvert.c -o /opt/mwm/160113/intermediate_data/osmconvert
/tmp/ccPJtLV8.o: In function `read_input':
osmconvert.c:(.text+0x4b3): undefined reference to `gzread'
/tmp/ccPJtLV8.o: In function `read_switch':
osmconvert.c:(.text+0x1fff): undefined reference to `gzread'
/tmp/ccPJtLV8.o: In function `oo__end':
osmconvert.c:(.text+0x21a3): undefined reference to `gzread'
osmconvert.c:(.text+0x2210): undefined reference to `gzclose'
/tmp/ccPJtLV8.o: In function `oo_open':
osmconvert.c:(.text.unlikely+0x37c8): undefined reference to `gzdopen'
osmconvert.c:(.text.unlikely+0x37df): undefined reference to `gzbuffer'
osmconvert.c:(.text.unlikely+0x37c8): undefined reference to `gzdopen' 
osmconvert.c:(.text.unlikely+0x37df): undefined reference to `gzbuffer'
osmconvert.c:(.text.unlikely+0x37fa): undefined reference to `gzopen64'
osmconvert.c:(.text.unlikely+0x382b): undefined reference to `gzbuffer'
/tmp/ccPJtLV8.o: In function `pb_input.part.42':
osmconvert.c:(.text.unlikely+0x724a): undefined reference to `inflateInit_'
osmconvert.c:(.text.unlikely+0x7282): undefined reference to `inflate'
osmconvert.c:(.text.unlikely+0x7293): undefined reference to `inflateEnd'
osmconvert.c:(.text.unlikely+0x729f): undefined reference to `inflateEnd'
/tmp/ccPJtLV8.o: In function `pw__obj_compress':
osmconvert.c:(.text.unlikely+0x75bf): undefined reference to `deflateInit_'
osmconvert.c:(.text.unlikely+0x75eb): undefined reference to `deflate'
osmconvert.c:(.text.unlikely+0x75fa): undefined reference to `deflateEnd'
osmconvert.c:(.text.unlikely+0x760e): undefined reference to `deflateEnd'
/tmp/ccPJtLV8.o: In function `oo_main':
osmconvert.c:(.text.unlikely+0xa4b2): undefined reference to `gzseek64'
osmconvert.c:(.text.unlikely+0xa68c): undefined reference to `gzseek64'
/tmp/ccPJtLV8.o: In function `main':
osmconvert.c:(.text.startup+0x7df): undefined reference to `zlibCompileFlags'
collect2: error: ld returned 1 exit status
```

@Zverik

> On 13.1.2016, at 10.20, RustamGimadiev <notifications@github.com> wrote:
> 
> В файле omim/tools/unix/generate_planetsh нужно
> 
>  [ ! -x "$OSMCTOOLS/osmconvert" ] && cc -x c -lz -O3 "$OMIM_PATH/tools/osmctools/osmconvertc" -o "$OSMCTOOLS/osmconvert" 
> заменить на
> 
> [ ! -x "$OSMCTOOLS/osmconvert" ] && cc -x c -O3 "$OMIM_PATH/tools/osmctools/osmconvertc" -o "$OSMCTOOLS/osmconvert" -lz
> так как ld опции при сборке должны идти в конце
> 
> Иначе будет ошибка:
> 
> + cc -x c -lz -O3 /omim/tools/osmctools/osmconvertc -o /opt/mwm/160113/intermediate_data/osmconvert
> /tmp/ccPJtLV8o: In function `read_input':
> osmconvertc:(text+0x4b3): undefined reference to `gzread'
> /tmp/ccPJtLV8o: In function `read_switch':
> osmconvertc:(text+0x1fff): undefined reference to `gzread'
> /tmp/ccPJtLV8o: In function `oo__end':
> osmconvertc:(text+0x21a3): undefined reference to `gzread'
> osmconvertc:(text+0x2210): undefined reference to `gzclose'
> /tmp/ccPJtLV8o: In function `oo_open':
> osmconvertc:(textunlikely+0x37c8): undefined reference to `gzdopen'
> osmconvertc:(textunlikely+0x37df): undefined reference to `gzbuffer'
> osmconvertc:(textunlikely+0x37c8): undefined reference to `gzdopen' 
> osmconvertc:(textunlikely+0x37df): undefined reference to `gzbuffer'
> osmconvertc:(textunlikely+0x37fa): undefined reference to `gzopen64'
> osmconvertc:(textunlikely+0x382b): undefined reference to `gzbuffer'
> /tmp/ccPJtLV8o: In function `pb_inputpart42':
> osmconvertc:(textunlikely+0x724a): undefined reference to `inflateInit_'
> osmconvertc:(textunlikely+0x7282): undefined reference to `inflate'
> osmconvertc:(textunlikely+0x7293): undefined reference to `inflateEnd'
> osmconvertc:(textunlikely+0x729f): undefined reference to `inflateEnd'
> /tmp/ccPJtLV8o: In function `pw__obj_compress':
> osmconvertc:(textunlikely+0x75bf): undefined reference to `deflateInit_'
> osmconvertc:(textunlikely+0x75eb): undefined reference to `deflate'
> osmconvertc:(textunlikely+0x75fa): undefined reference to `deflateEnd'
> osmconvertc:(textunlikely+0x760e): undefined reference to `deflateEnd'
> /tmp/ccPJtLV8o: In function `oo_main':
> osmconvertc:(textunlikely+0xa4b2): undefined reference to `gzseek64'
> osmconvertc:(textunlikely+0xa68c): undefined reference to `gzseek64'
> /tmp/ccPJtLV8o: In function `main':
> osmconvertc:(textstartup+0x7df): undefined reference to `zlibCompileFlags'
> collect2: error: ld returned 1 exit status
> —
> Reply to this email directly or view it on GitHub.
>

--- a/tools/unix/generate_planet.sh
+++ b/tools/unix/generate_planet.sh
@@ -220,9 +220,9 @@ fi
 
 if [ "$MODE" == "coast" ]; then
   putmode
-  [ ! -x "$OSMCTOOLS/osmconvert" ] && cc -x c -lz -O3 "$OMIM_PATH/tools/osmctools/osmconvert.c" -o "$OSMCTOOLS/osmconvert"
-  [ ! -x "$OSMCTOOLS/osmupdate"  ] && cc -x c         "$OMIM_PATH/tools/osmctools/osmupdate.c"  -o "$OSMCTOOLS/osmupdate"
-  [ ! -x "$OSMCTOOLS/osmfilter"  ] && cc -x c     -O3 "$OMIM_PATH/tools/osmctools/osmfilter.c"  -o "$OSMCTOOLS/osmfilter"
+  [ ! -x "$OSMCTOOLS/osmconvert" ] && cc -x c -O3 "$OMIM_PATH/tools/osmctools/osmconvert.c" -o "$OSMCTOOLS/osmconvert" -lz
+  [ ! -x "$OSMCTOOLS/osmupdate"  ] && cc -x c     "$OMIM_PATH/tools/osmctools/osmupdate.c"  -o "$OSMCTOOLS/osmupdate"
+  [ ! -x "$OSMCTOOLS/osmfilter"  ] && cc -x c -O3 "$OMIM_PATH/tools/osmctools/osmfilter.c"  -o "$OSMCTOOLS/osmfilter"
   if [ -n "$OPT_DOWNLOAD" ]; then
     # Planet download is requested
     log "STATUS" "Step 0: Downloading and converting the planet"

--- a/tools/unix/generate_planet_routing.sh
+++ b/tools/unix/generate_planet_routing.sh
@@ -39,7 +39,7 @@ if [ "$1" == "pbf" ]; then
   PLANET="${PLANET:-$HOME/planet/planet-latest.o5m}"
   OSMCTOOLS="${OSMCTOOLS:-$HOME/osmctools}"
   [ ! -d "$OSMCTOOLS" ] && OSMCTOOLS="$INTDIR"
-  [ ! -x "$OSMCTOOLS/osmconvert" ] && cc -x c -lz -O3 "$OMIM_PATH/tools/osmctools/osmconvert.c" -o "$OSMCTOOLS/osmconvert"
+  [ ! -x "$OSMCTOOLS/osmconvert" ] && cc -x c -O3 "$OMIM_PATH/tools/osmctools/osmconvert.c" -o "$OSMCTOOLS/osmconvert" -lz
 
   TMPBORDERS="$INTDIR/tmpborders"
   [ -d "$TMPBORDERS" ] && rm -r "$TMPBORDERS"

